### ISSUE TITLE
Longform QA demo breaks after clearing cache

### ISSUE THREAD

**Issue by ontocord** [huggingface/transformers#8830]
## Environment info

Browser: Chrome, running on windows.

Running demo at: http://35.226.96.115:8080/

Linked from https://github.com/huggingface/transformers/tree/master/examples/longform-qa

### Who can help
@sgugger
 
## Information
I had clicked "Clear Cache" in the app and when I did another search errors came up in the browers. subsequent runs also produces errors in the browser.

RuntimeError: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 8987501056 bytes (error 2 out of memory)
Traceback:
File "/home/yacine/anaconda3/lib/python3.7/site-packages/streamlit/ScriptRunner.py", line 322, in _run_script
    exec(code, module.__dict__)
File "/home/yacine/Code/transformers/examples/longform-qa/eli5_app.py", line 78, in <module>
    passages, gpu_dense_index, es_client = load_indexes()
File "/home/yacine/anaconda3/lib/python3.7/site-packages/streamlit/caching.py", line 591, in wrapped_func
    return get_or_create_cached_value()
File "/home/yacine/anaconda3/lib/python3.7/site-packages/streamlit/caching.py", line 575, in get_or_create_cached_value
    return_value = func(*args, **kwargs)
File "/home/yacine/Code/transformers/examples/longform-qa/eli5_app.py", line 58, in load_indexes
    wiki40b_gpu_index_flat.add(wiki40b_passage_reps)  # TODO fix for larger GPU
File "/home/yacine/anaconda3/lib/python3.7/site-packages/faiss/__init__.py", line 138, in replacement_add
    self.add_c(n, swig_ptr(x))
File "/home/yacine/anaconda3/lib/python3.7/site-packages/faiss/swigfaiss.py", line 4245, in add
    return _swigfaiss.GpuIndexFlat_add(self, arg2, x)
## Expected behavior

Expected to have the results of the ELI5 search

**Comment by LysandreJik** [comment]
This seems to be an out-of-memory error! @yjernite might know what's up.
